doc/source/concept_and_workflow/shell_scripts.rst

```diff
@@ -204,9 +205,28 @@ suseSetupProductInformation
   and installs all product specific packages. This function fails
   when :command:`zypper` is not the appliances package manager.
 
+baseVagrantSetup
+  Configures the image to work as a vagrant box by performing the following
+  changes:
+  - add the ``vagrant`` user to :file:`/etc/sudoers` or
+    :file:`/etc/sudoers.d/vagrant`
+  - insert the insecure vagrant ssh key, apply recommended ssh settings and
+    start the ssh daemon
+  - create the default shared folder :file:`/vagrant`
+
+suseVagrantSetup
+  Additionally to ``baseVagrantSetup``, this function also performs the
+  following actions:
+  - if the package ``virtualbox-guest-tools`` is installed, load the
+    ``vboxsf`` module
+  - remove all udev rules for networking that match
+    :file:`/etc/udev/rules.d/*-net.rules` so that the wired network is called
+    ``eth0`` if the kiwi profile is not called ``virtualbox``
+  - setup DHCP on ``eth0``
```

Comment:
Why is `suseVagrantSetup` needed? Why isn't the package providing this setup like the Fedora `virtualbox-guest-additions` package does?

Reply:
At least the udev rules removal & dhcp config is really only applicable for vagrant boxes (but it's distro specific, I have not checked if this would work in Fedora).

My guess concerning the `virtualbox-guest-additions` package is, that not everyone who installs virtualbox, wants to have the `vboxsf` kernel module.

Reply:
> that not everyone who installs virtualbox, wants to have the vboxsf kernel module.

I think if the image description has provider="virtualbox" virtualbox_guest_additions_present="true" set and vboxsf kernel module exists it is wanted. The kiwi image description is available below /image when your script runs. It would be possible to check base on those values to come to a decision

> At least the udev rules removal 

This should not be done imho

> dhcp config 

If you consider to put that code into it's own method you could make only the dhcp setup distro specific.
Regarding that network setup I'm not sure if we want to provide code to do it. The network setup is very
different between all distributions and I would rather document that step instead of writing extra code.
Alternatively we could utilize existing tools to create the network setup if possible.

In the end I think suseVagrantSetup should be avoided and we should aim for a generic method as much
as possible. Just my2cents